docs/pages/api/telemetry/index.ts

```diff
@@ -0,0 +1,38 @@
+import { Analytics } from "@segment/analytics-node";
+import { NextApiRequest, NextApiResponse } from "next";
+
+export default async function forwardTelemetry(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method !== "POST") {
+    res.setHeader("Allow", ["POST"]);
+    return res.status(405).end(`Method ${req.method} Not Allowed`);
+  }
+
+  if (!process.env.SEGMENT_WRITE_KEY) {
+    console.error('Missing required environment variable "SEGMENT_WRITE_KEY".');
+    return res
+      .status(500)
+      .json({ error: "An error occurred while processing telemetry data." });
+  }
+
+  try {
+    const analytics = new Analytics({
+      writeKey: process.env.SEGMENT_WRITE_KEY,
+    });
```

Comment:
I wonder if we can do this stuff above, outside the function scope.

Reply:
no :/ - I tried doing it outside of the function but Typescript complains.

this is what chatgpt says (although I'm not exactly sure this is true):

> In Node.js, environment variables are usually set when the process starts and are available for the duration of the process. However, in a serverless or "edge" environment (like the one you're running your Next.js code in), individual requests are handled in isolation and might not share the same environment variables state.